includes/wc-payment-api/class-wc-payments-http.php

```diff
@@ -71,6 +71,15 @@ public function remote_request( $args, $body = null, $is_site_specific = true, $
 			$args['url'] = implode( '?', $url );
 		}
 
+		// Generate Idempotency key for PATCH, POST and PUT requests.
+		if (
```

Comment:
This check for request methods for adding the idempotency key is new, right? Maybe we could only refactor in this PR without any functionality changes to isolate any side effects?

Reply:
Yes, this checks for idempotency key and if it doesn't exist, it will create a new one.
The problem is that the test will fail on PHP 8 version if we leave UUID generation as it was. The only solution I found so far is to remove that logic from payments api client to wc payments HTTP client because idempotency key is required  there.

Reply:
Makes sense but do we need the check for specific request methods here for this to work? I don't see a similar check in the existing flow so maybe we could move it here without adding this conditional check.

Reply:
I followed the original concept and added check for specific request methods, as can be seen here:

https://github.com/Automattic/woocommerce-payments/blob/0fd92cf70a3d02e24f6313ad73fef26fa336fc9b/includes/wc-payment-api/class-wc-payments-api-client.php#L1596

Reply:
Ah, my bad, missed that somehow. 😅